File: 2022/1_Calorie_Counting/part_2.ts
import { readFileSync } from 'fs'

const input: number[][] = readFileSync('input.txt', 'utf-8')
  .split('\n\n')
  .map((elfPack) => elfPack.split('\n').map((candyBar) => parseInt(candyBar)))

// --------------

const bagsWithTotalCallories = input.map((bag) =>
  bag.reduce((sum, number) => sum + number)
)

const sortedBags = bagsWithTotalCallories.sort((a, b) => b - a)

function sumOfTopBags(numberOfBags: number) {
  let sumOfBags = 0
  for (let i = 0; i < numberOfBags; i++) {
    sumOfBags += sortedBags[i]
  }
  return sumOfBags
}

console.log(sumOfTopBags(3))

File: 2022/1_Calorie_Counting/part_2.rb
input =
  File
    .open('input.txt')
    .read
    .split("\n\n")
    .map { |bag| bag.split("\n").map { |candy| candy.to_i } }

@sorted_bag_totals =
  input.map { |bag| bag.reduce(0) { |sum, number| sum + number } }.sort.reverse

def sumOfBags(number)
  sum_of_number_of_bags = 0
  number.times { |i| sum_of_number_of_bags += @sorted_bag_totals[i] }
  return sum_of_number_of_bags
end

print sumOfBags(3)
